packages/wdio-cli/src/launcher.ts

```diff
@@ -244,6 +246,12 @@ class Launcher {
              * fail if no specs were found or specified
              */
             if (Object.values(this._schedule).reduce((specCnt, schedule) => specCnt + schedule.specs.length, 0) === 0) {
+                const { total, current } = config.shard
+                if (total > 1) {
+                    log.info(`No specs to execute in shard ${current}/${total}, exiting!`)
+                    return resolve(0)
```

Comment:
If we exit with code 1 when there are no tests, I would expect the same here

Reply:
For sharding, this would be not desireable. For example: let's say a user has a CI setup that by defaults uses 10 workers and calls wdio always with `--shard 1/10` ... `--shard 10/10`. Now let's say we have 100 test files and provide a dynamic parameter like `--spec="SOME_PATTERN"` which can resolves in 9 test files. In this case just because the dynamic spec parameter causes let's files to be selected than for what sharding is set up for should not cause the worker to break.